sale.py

```diff
@@ -352,6 +353,10 @@ def proceed(cls, sales):
         super(Sale, cls).proceed(sales)
 
         for sale in sales:
+            if sale.payment_processing_state:
```

Comment:
This is not where this check should happen. This should happen in capture and authorize methods.